pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go

```diff
@@ -18,104 +18,42 @@ package noderesources
 
 import (
 	"context"
-	"reflect"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
-	"github.com/google/go-cmp/cmp/cmpopts"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
-	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	plfeature "k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
 	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
 	"k8s.io/kubernetes/pkg/scheduler/internal/cache"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
 )
 
-func TestNodeResourcesLeastAllocated(t *testing.T) {
-	labels1 := map[string]string{
-		"foo": "bar",
-		"baz": "blah",
-	}
-	labels2 := map[string]string{
-		"bar": "foo",
-		"baz": "blah",
-	}
-	machine1Spec := v1.PodSpec{
-		NodeName: "machine1",
-	}
-	machine2Spec := v1.PodSpec{
-		NodeName: "machine2",
-	}
-	noResources := v1.PodSpec{
-		Containers: []v1.Container{},
-	}
-	cpuOnly := v1.PodSpec{
-		NodeName: "machine1",
-		Containers: []v1.Container{
-			{
-				Resources: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU:    resource.MustParse("1000m"),
-						v1.ResourceMemory: resource.MustParse("0"),
-					},
-				},
-			},
-			{
-				Resources: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU:    resource.MustParse("2000m"),
-						v1.ResourceMemory: resource.MustParse("0"),
-					},
-				},
-			},
-		},
-	}
-	cpuOnly2 := cpuOnly
-	cpuOnly2.NodeName = "machine2"
-	cpuAndMemory := v1.PodSpec{
-		NodeName: "machine2",
-		Containers: []v1.Container{
-			{
-				Resources: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU:    resource.MustParse("1000m"),
-						v1.ResourceMemory: resource.MustParse("2000"),
-					},
-				},
-			},
-			{
-				Resources: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU:    resource.MustParse("2000m"),
-						v1.ResourceMemory: resource.MustParse("3000"),
-					},
-				},
-			},
-		},
-	}
-	defaultResourceLeastAllocatedSet := []config.ResourceSpec{
+func TestLeastAllocatedScoringStrategy(t *testing.T) {
+	defaultResources := []config.ResourceSpec{
 		{Name: string(v1.ResourceCPU), Weight: 1},
 		{Name: string(v1.ResourceMemory), Weight: 1},
 	}
+
 	extendedRes := "abc.com/xyz"
 	extendedResourceLeastAllocatedSet := []config.ResourceSpec{
 		{Name: string(v1.ResourceCPU), Weight: 1},
 		{Name: string(v1.ResourceMemory), Weight: 1},
 		{Name: extendedRes, Weight: 1},
 	}
-	cpuMemoryAndExtendedRes := *cpuAndMemory.DeepCopy()
-	cpuMemoryAndExtendedRes.Containers[0].Resources.Requests[v1.ResourceName(extendedRes)] = resource.MustParse("2")
+
+	noResourcePod := st.MakePod().Pod
```

Comment:
Let's just instantiate it in place inside all sub-tests:

```go
{
    ...
    requestedPod: st.MakePod().Obj(),
}
```

Reply:
done :)